File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        left, right = 0, 1
        ans = 0
        while right < len(colors):
            sum_ = neededTime[left]
            mx_ = neededTime[left]
            while right < len(colors) and colors[left] == colors[right]:
                sum_ += neededTime[right]
                mx_ = max(mx_, neededTime[right])
                right += 1
            
            if sum_ != mx_:
                ans += sum_-mx_
            
            left = right
            right +=1
        
        return ans
        

File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.cpp
#include <algorithm>

class Solution {
public:
    int minCost(string colors, vector<int>& neededTime) {
        int left = 0, right = 1;
        int ans = 0;
        
        while (right < neededTime.size()){
            int sum = neededTime[left];
            int mx = neededTime[left];
            while (right < neededTime.size() && colors[left] == colors[right]){
                sum += neededTime[right];
                mx = max(mx, neededTime[right]);
                right += 1;
                
            }
            
            if (mx != sum){
                ans += sum-mx;
            }
            
            left = right;
            right += 1;
        }
        return ans;
    }
};